### PR TITLE
Rewrite [codacy] badges; rm unused svg helpers

### PR DESCRIPTION
**Pull request by paulmelnikow**

### CODE REVIEW

File: services/codacy/codacy-grade.tester.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  )
+
+t.create('Code qualiy on branch')
+  .get('/grade/e27821fb6289410b8f58338c7e0bc686/master.json')
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'code quality',
+      value: isGrade,
+    })
+  )
```
Comment (paulmelnikow) [inline services/codacy/codacy-grade.tester.js]: We should think about what tests should be added here.

**Comment by paulmelnikow** [comment]
This pull request **introduces 5 alerts** when merging c6763c5f460105e1770d12b71ff2139a01a5f735 into c8ce4fabb4812f0e56d7e6e1ca00d479b062c449 - [view on LGTM.com](https://lgtm.com/projects/g/badges/shields/rev/pr-71066f6bf7e81cbc9c5ce430efa639acd3ad8bf7)

**new alerts:**

* 4 for Wrong use of &#39;this&#39; for static method
* 1 for Incomplete string escaping or encoding

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by paulmelnikow** [comment]
This pull request **introduces 1 alert** when merging e92cb2278dc4637cdd9063e8501a83ca2ef15ee2 into 88402dd7a8ed64953937157de866d9a62dab8c48 - [view on LGTM.com](https://lgtm.com/projects/g/badges/shields/rev/pr-82a614b11113ce27e1a595b0b5185581fb4241ef)

**new alerts:**

* 1 for Incomplete string escaping or encoding

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: services/codacy/codacy-coverage.service.js [hunk 2]
```diff
@@ -1,78 +1,82 @@
...
+  async handle({ projectId, branch }) {
+    const { message: coverageString } = await this._requestSvg({
+      schema,
+      url: `https://api.codacy.com/project/badge/coverage/${encodeURIComponent(
+        projectId
+      )}`,
+      options: { qs: { branch } },
+      valueMatcher: /text-anchor="middle">([^<>]+)<\/text>/,
+    })
+    if (coverageString === '!') {
```
Comment (PyvesB) [inline services/codacy/codacy-coverage.service.js]: I don't think "not found" is the correct meaning of the exclamation mark. Consider project `e27821fb6289410b8f58338c7e0bc686`:
- coverage:  (https://api.codacy.com/project/badge/coverage/e27821fb6289410b8f58338c7e0bc686)
- grade:  (https://api.codacy.com/project/badge/grade/e27821fb6289410b8f58338c7e0bc686)

The project is found and as far as I can tell, `!` is equivalent to "coverage unknown" or "coverage  not available for this project". When a project is not found a 404 is returned.

File: services/codacy/codacy-grade.tester.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    value: 'project or branch not found',
+  })
+
+// This is a known bug. The badge endpoint for a nonexistent branch returns
+// the same result. It seems possible the branch specification isn't being
+// considered at all.
+// e.g.
+// https://api.codacy.com/project/badge/grade/e27821fb6289410b8f58338c7e0bc686
+// https://api.codacy.com/project/badge/grade/e27821fb6289410b8f58338c7e0bc686?branch=foo
+// t.create('Code quality on branch (branch not found)')
```
Comment (PyvesB) [inline services/codacy/codacy-grade.tester.js]: We've been following the [documentation](https://support.codacy.com/hc/en-us/articles/212799365-Badges) regarding the branch parameter, either it's out of date or the feature isn't working properly. @nemchik has already contacted the support team in the past, maybe it's worth him enquiring about this parameter? 😄

Comment (paulmelnikow) [reply services/codacy/codacy-grade.tester.js]: Ah, right. I reached out to them on Twitter a few days back. Their response was the branch parameter is used if it matches a branch, and it's ignored if it doesn't match a branch, i.e. it fails silently. I'll update the comment.

File: services/codacy/codacy-coverage.service.js (hunk 2 above)
Comment (paulmelnikow) [reply services/codacy/codacy-coverage.service.js]: Thanks for digging into that. I'll see if I can't get a separate "project not found" test case working here.

**Pull request merged**